platform/osx/display_server_osx.mm

```diff
@@ -3524,7 +3524,13 @@ static void keyboard_layout_changed(CFNotificationCenterRef center, void *observ
 	}
 #endif
 
-	WindowID main_window = _create_window(p_mode, Rect2i(Point2i(), p_resolution));
+	float newDisplayScale = OS_OSX::get_singleton()->is_hidpi_allowed() ? newBackingScaleFactor : 1.0;
+
+	const NSRect contentRect = [wd.window_view frame];
+	Point2i window_position(
+			(contentRect.size.width * newDisplayScale - p_resolution.x) / 2,
+			(contentRect.size.height * newDisplayScale - p_resolution.y) / 2);
+	WindowID main_window = _create_window(p_mode, Rect2i(window_position, p_resolution));
```

Comment:
WindowData (`wd`) is created inside `_create_window` and it's not available here, neither is `main_window` on Linux. 

And it should be `sceen_size - p_resolution` not `window_size - p_resolution`, cause `p_resolution` is the window size in this case.

Probably something like:
```c++
Point2i window_position(
		(screen_get_size(0).x - p_resolution.x) / 2,
		(screen_get_size(0).y - p_resolution.y) / 2);
```
on both.

Reply:
sorry for that, the visual studio intelisence  not working on osx file, no auto compilation and every text grayed out. @bruvzg  now I've ammend the commit

Reply:
I've changed 0 to window_get_current_screen(), and the windows display server position also need to changed like this (because of DPI value) and I'll do it ASAP @bruvzg

Reply:
I do not think `window_get_current_screen()` will work here, there's no any window created, 
it will return `-1` and crash, `0` should be OK for the primary screen.

Reply:
And won't work without argument either, it's the same as `window_get_current_screen()` but `window_get_current_screen()` is called from the `screen_get_size`.

Reply:
Oh, I reset the screen to 0 but does it always be 0 (isn't there any enum or something?)

Reply:
It's just sequential display number, AFAIK primary display should be always `0`. 
Enum only have one value - `SCREEN_OF_MAIN_WINDOW = -1`, which means, use `window_get_current_screen()` to get display number.